lib/tests/unit/modules/apply_test.cc

```diff
@@ -0,0 +1,108 @@
+#include "root_path.hpp"
+#include "../../common/content_format.hpp"
+
+#include <pxp-agent/modules/apply.hpp>
+#include <pxp-agent/configuration.hpp>
+#include <pxp-agent/util/process.hpp>
+
+#include <cpp-pcp-client/protocol/chunks.hpp>       // ParsedChunks
+
+#include <leatherman/json_container/json_container.hpp>
+#include <leatherman/util/scope_exit.hpp>
+#include <leatherman/file_util/file.hpp>
+
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/erase.hpp>
+#include <boost/date_time/posix_time/posix_time_types.hpp>
+
+#include <catch.hpp>
+
+#include <string>
+#include <vector>
+#include <unistd.h>
+
+using namespace PXPAgent;
+
+namespace fs = boost::filesystem;
+namespace pt = boost::posix_time;
+namespace lth_jc = leatherman::json_container;
+namespace lth_util = leatherman::util;
+namespace lth_file = leatherman::file_util;
+
+static const std::string SPOOL_DIR { std::string { PXP_AGENT_ROOT_PATH }
+                                     + "/lib/tests/resources/test_spool" };
+
+static const auto STORAGE = std::make_shared<ResultsStorage>(SPOOL_DIR, "0d");
+
+static const std::string CACHE_DIR { std::string { PXP_AGENT_ROOT_PATH }
+                                          + "/lib/tests/resources/tasks-cache" };
+
+// Disable cache ttl so we don't delete fixtures.
+static const std::string CACHE_TTL { "0d" };
+
+static const auto MODULE_CACHE_DIR = std::make_shared<ModuleCacheDir>(CACHE_DIR, CACHE_TTL);
+
+static const std::string TEMP_CACHE_DIR { CACHE_DIR + "/temp" };
+
+static const std::vector<std::string> MASTER_URIS { { "https://_master1", "https://_master2", "https://_master3" } };
+
+static const std::string CA { "mock_ca" };
+
+static const std::string CRT { "mock_crt" };
+
+static const std::string KEY { "mock_key" };
+
+TEST_CASE("Modules::Apply", "[modules]") {
+    SECTION("can successfully instantiate") {
+        REQUIRE_NOTHROW(Modules::Apply(PXP_AGENT_BIN_PATH, MASTER_URIS, CA, CRT, KEY, "", 10, 20, MODULE_CACHE_DIR, STORAGE));
+    }
+}
+
+TEST_CASE("Modules::Apply::hasAction", "[modules]") {
+    Modules::Apply mod { PXP_AGENT_BIN_PATH, MASTER_URIS, CA, CRT, KEY, "", 10, 20, MODULE_CACHE_DIR, STORAGE };
+    SECTION("correctly reports false") {
+        REQUIRE(!mod.hasAction("foo"));
+    }
+
+    SECTION("correctly reports true") {
+        REQUIRE(mod.hasAction("apply"));
+    }
+}
```

Comment:
Given we do not have an agent (and ruby interpreter) for acceptance tests we cant really do much more than this super minimal set of unit tests. Unless anybody has a better idea? I thought about putting some bash in a $tasks-cache/apply_ruby_shim/apply_ruby_shim.rb file with a bash shebang that echod back the input, BUT that would never work on windows (and is not a meaningful test).